src/webui/api/searchcontroller.cpp

```diff
@@ -143,35 +125,33 @@ void SearchController::stopAction()
     requireParams({u"id"_qs});
 
     const int id = params()[u"id"_qs].toInt();
-    ISession *const session = sessionManager()->session();
 
-    const auto searchHandlers = session->getData<SearchHandlerDict>(SEARCH_HANDLERS);
-    if (!searchHandlers.contains(id))
+    const auto iter = m_searchHandlers.find(id);
+    if (iter == m_searchHandlers.end())
         throw APIError(APIErrorType::NotFound);
 
-    const SearchHandlerPtr searchHandler = searchHandlers[id];
+    const std::shared_ptr<SearchHandler> searchHandler = iter.value();
 
     if (searchHandler->isActive())
     {
         searchHandler->cancelSearch();
-        removeActiveSearch(session, id);
+        m_activeSearches.remove(id);
     }
 }
 
 void SearchController::statusAction()
 {
     const int id = params()[u"id"_qs].toInt();
 
-    const auto searchHandlers = sessionManager()->session()->getData<SearchHandlerDict>(SEARCH_HANDLERS);
-    if ((id != 0) && !searchHandlers.contains(id))
+    if ((id != 0) && !m_searchHandlers.contains(id))
         throw APIError(APIErrorType::NotFound);
 
     QJsonArray statusArray;
-    const QList<int> searchIds {(id == 0) ? searchHandlers.keys() : QList<int> {id}};
+    const QList<int> searchIds {(id == 0) ? m_searchHandlers.keys() : QList<int> {id}};
 
     for (const int searchId : searchIds)
     {
-        const SearchHandlerPtr searchHandler = searchHandlers[searchId];
+        const std::shared_ptr<SearchHandler> searchHandler = m_searchHandlers[searchId];
```

Comment:
ditto.

Reply:
> ditto.

No. Look for yourself - there is no double lookup here, so there is no point in complicating it.

Reply:
```suggestion
        const std::shared_ptr<SearchHandler> &searchHandler = m_searchHandlers[searchId];
```

Reply:
Done.